aexpect/remote.py

```diff
@@ -826,6 +826,68 @@ def stop_server(session):
         d_session.close()
 
 
+def login_reusing_session(session, log_filename=None, log_function=None,
```

Comment:
This is a bit misleading name. At first glance I thought you reuse the `session` connection, but with scp it became clear that is not the case. How about `login_based_on_session` or `login_from_session` (`scp_from|to_session`)?

Reply:
Yes, makes perfect sense. I was doing too many other things when naming these functions. Resolved in the next push.